pacsman/pynetdicom_client.py

```diff
@@ -0,0 +1,399 @@
+            dataset.BodyPartExamined = ''
+            dataset.SeriesDescription = ''
+            dataset.SeriesDate = ''
+            dataset.SeriesTime = ''
+            dataset.SeriesInstanceUID = ''
+            dataset.PatientPosition = ''
+            dataset.QueryRetrieveLevel = 'SERIES'
+
+            responses = assoc.send_c_find(dataset, query_model='S')
+
+            series_infos = []
+            for (status, series) in responses:
+                logger.debug(status)
+                logger.debug(series)
+
+                if status.Status not in status_success_or_pending:
+                    raise Exception('Series C-FIND Failure Response: 0x{0:04x}'.format(status.Status))
+
+                if series and (modality_filter is None or
+                               getattr(series, 'Modality', '') in modality_filter):
+                    description = getattr(series, 'SeriesDescription', '')
+                    body_part_examined = getattr(series, 'BodyPartExamined', None)
+                    if body_part_examined:
+                        description += f' ({body_part_examined})'
+
+                    with association(ae, self.pacs_url, self.pacs_port) as series_assoc:
+                        series_dataset = Dataset()
+                        series_dataset.SeriesInstanceUID = series.SeriesInstanceUID
+                        series_dataset.QueryRetrieveLevel = 'IMAGE'
+                        series_dataset.SOPInstanceUID = ''
+
+                        series_responses = series_assoc.send_c_find(series_dataset, query_model='S')
+                        image_ids = []
+                        for (instance_status, instance) in series_responses:
+                            logger.debug(instance)
+                            if instance_status.Status in status_success_or_pending:
+                                if hasattr(instance, 'SOPInstanceUID'):
+                                    image_ids.append(instance.SOPInstanceUID)
+                            else:
+                                raise Exception(
+                                    'Image C-FIND Failure Response: 0x{0:04x}'.format(
+                                        status.Status))
+
+                    info = SeriesInfo(series_id=series.SeriesInstanceUID, description=description,
+                                      modality=series.Modality, num_images=len(image_ids),
+                                      acquisition_datetime=series.SeriesDate)
+
+                    series_infos.append(info)
+
+        return series_infos
+
+    def fetch_images_as_files(self, series_id):
+        """
+        Fetches series images from PACS with C-MOVE/C-STORE
+        :param series_id: SeriesInstanceUID from PACS
+        :return: a path to a directory full of dicom files on success, None if not found
+        """
+        series_path = os.path.join(self.dicom_dir, series_id)
+        scp = StorageSCP(self.client_ae, series_path)
```

Comment:
Perhaps we could make a context manager for the `scp` construction/cleanup?  It looks like the code is repeated a few times.